### PR TITLE
Feat/serverless

### PR DESCRIPTION
**Pull request by jackey8616**
Preparation for implement [Middy](https://middy.js.org).

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [x] Have you written unit tests?
- [x] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [x] This PR is associated with an existing issue?

#1139 
#1445

### CODE REVIEW

File: packages/runtime/src/routeGeneration/templates/serverless/serverlessTemplateService.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      throw new ValidateError(fieldErrors, '');
+    }
+    return values;
+  }
+
+  protected returnHandler(params: ServerlessReturnHandlerParameters) {
+    const { statusCode, data } = params;
+    let { headers } = params;
+    headers = headers || {};
+    /*
```
Comment (WoH) [inline packages/runtime/src/routeGeneration/templates/serverless/serverlessTemplateService.ts:92]: .

File: tests/integration/express-server.spec.ts [hunk 2]
```diff
@@ -24,395 +24,409 @@ import {
 const basePath = '/v1';
 
 describe('Express Server', () => {
-  it('can handle get request to root controller`s path', () => {
-    return verifyGetRequest(basePath + '/', (_err, res) => {
-      const model = res.body as TestModel;
-      expect(model.id).to.equal(1);
+  describe('RootController', () => {
```
Comment (WoH) [inline tests/integration/express-server.spec.ts:27]: Even with whitespace hidden, the changes in this file are hard to verify. Why do we need to change the express server specs in this PR?

File: packages/cli/package.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
   ],
   "scripts": {
     "build": "yarn clean && yarn tsc && yarn copy-templates",
-    "copy-templates": "copyfiles -u 1 ./src/routeGeneration/templates/*.hbs ./dist",
+    "copy-templates": "copyfiles -u 1 ./src/routeGeneration/templates/*.hbs ./src/routeGeneration/templates/**/*.hbs ./dist",
```
[the remark this answers is not in the document]
Comment (jackey8616) [reply packages/cli/package.json:25]: hbs template all put in `src/routeGeneration/templates` folder.
But serverless template are put in another subfolder which is `src/routeGeneration/templates/serverless`.

Any suggestion?

File: tests/integration/express-server.spec.ts (hunk 2 above)
Comment (jackey8616) [reply tests/integration/express-server.spec.ts:27]: When I cover whole tests in express-server.spec, I also grouped these test cases with different describe.

**Pull request closed without merging**